sql/txn.go

```diff
@@ -50,7 +50,7 @@ func (p *planner) SetTransaction(n *parser.SetTransaction) (planNode, error) {
 func (p *planner) setIsolationLevel(level parser.IsolationLevel) error {
 	switch level {
 	case parser.UnspecifiedIsolation:
-		return p.txn.SetIsolation(p.session.DefaultIsolationLevel)
+		return nil
```

Comment:
`sql.Executor` creates a new `client.Txn` by calling `Executor.newTxn` which calls `Executor.initializeTxn` in turn.  `Executor.initializeTxn` sets default isolation level. So calling `p.txn.SetIsolation(p.session.DefaultIsolationLevel)` is not needed.

With `p.txn.SetIsolation(p.session.DefaultIsolationLevel)`, `SET TRANSACTION` with only user priority specified will change isolation level. For details, see the SQL statements added at the end of `sql/testdata/txn`.